src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder.Refiners
+{
+    public class ShellRefiner : CommonLanguageRefiner, ILanguageRefiner
+    {
+        public ShellRefiner(GenerationConfiguration configuration) : base(configuration) { }
+        public override void Refine(CodeNamespace generatedCode)
+        {
+            // Remove PathSegment field
+            // Convert Properties to AddCommand
+            AddDefaultImports(generatedCode, defaultUsingEvaluators);
+            MoveClassesWithNamespaceNamesUnderNamespace(generatedCode);
+            ConvertUnionTypesToWrapper(generatedCode, _configuration.UsesBackingStore);
+            AddPropertiesAndMethodTypesImports(generatedCode, false, false, false);
+            CreateCommandBuilders(generatedCode);
+            AddAsyncSuffix(generatedCode);
+            AddInnerClasses(generatedCode, false);
+            AddParsableInheritanceForModelClasses(generatedCode);
+            CapitalizeNamespacesFirstLetters(generatedCode);
+            ReplaceBinaryByNativeType(generatedCode, "Stream", "System.IO");
+            MakeEnumPropertiesNullable(generatedCode);
+            ReplaceReservedNames(generatedCode, new CSharpReservedNamesProvider(), x => $"@{x.ToFirstCharacterUpperCase()}");
+            DisambiguatePropertiesWithClassNames(generatedCode);
+            AddConstructorsForDefaultValues(generatedCode, false);
+            AddSerializationModulesImport(generatedCode);
+        }
+        private static void DisambiguatePropertiesWithClassNames(CodeElement currentElement)
+        {
```

Comment:
a number of the methods here are already defined in CSharp refiner, move them to the common language refiner instead and remove the duplicates from here please